File: r/anagram/R/anagram.R
# I don't understand why this isn't working. When I try the individual components in the R REPL, it seems to do the right thing. Yet, the tests fail. wat

sortString <- function(str) {
  chars <- strsplit(str, NULL)[[1]]
  return(paste(sort(chars), collapse=''))
}

anagram <- function(subject, candidates) {
  sortedSubject <- sortString(subject)
  return(Filter(function(candidate)
                  identical(sortString(candidate), sortedSubject),
                candidates))
}

File: clojure/anagram/src/anagram.clj
(ns anagram)

(defn anagram-of [word]
  (fn [candidate]
    (let [[w c] (map #(.toLowerCase %) [word candidate])]
      (and (not= w c) (= (sort w) (sort c))))))

(defn anagrams-for [word candidates]
  (filter (anagram-of word) candidates))
